### PR TITLE
Clean up kvutils logs

### PR DESCRIPTION
**Pull request by mziolekda**
- Clean up kvutils logs
- Convert config validation to be Committer-based

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/ConfigCommitter.scala [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+package com.daml.ledger.participant.state.kvutils.committer
+
+import com.daml.ledger.participant.state.kvutils.Conversions._
+import com.daml.ledger.participant.state.kvutils.DamlKvutils._
+import com.daml.ledger.participant.state.kvutils.committing.Common.getCurrentConfiguration
+import com.daml.ledger.participant.state.v1.Configuration
+
+private[kvutils] object ConfigCommitter {
+  case class Result(
+      builder: DamlConfigurationEntry.Builder,
```
Comment (dajmaki) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/ConfigCommitter.scala]: make this `DamlConfigurationSubmission` as we're not really using the builder to gradually build up the result.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala [hunk 2]
```diff
@@ -37,6 +38,9 @@ object KeyValueCommitting {
   def packDamlLogEntryId(entry: DamlLogEntryId): ByteString = entry.toByteString
   def unpackDamlLogEntryId(bytes: ByteString): DamlLogEntryId = DamlLogEntryId.parseFrom(bytes)
 
+  private def estimateMaximumRecordTime(recordTime: Timestamp): Timestamp =
```
Comment (dajmaki) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/KeyValueCommitting.scala]: please add comments on why this is temporarily here.

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/ConfigCommitter.scala [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    val rejections = metricsRegistry.counter("rejections")
+  }
+
+  private def rejectionTraceLog(
+      msg: String,
+      configurationEntry: DamlConfigurationEntry.Builder): Unit =
+    logger.trace(
+      s"Configuration rejected, $msg, correlationId=${configurationEntry.getSubmissionId}")
+
+  private val checkTtl: Step = (ctx, result) => {
```
Comment (dajmaki) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/ConfigCommitter.scala]: consider already in this PR pulling the checkTtl and authorizeSubmission to be built into Committer (we'll of course need to specify how to build the timeout/participant not authorized results).

Comment (mziolekda) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/ConfigCommitter.scala]: Haven't done that one. The commonalities are still too hard to spot through the thicket of diverging abstractions.

**Pull request merged**